### ISSUE TITLE
Support Application Default Credentials

### ISSUE THREAD

**Issue by zudov** [f-f/googlica#13]
https://developers.google.com/identity/protocols/application-default-credentials

**Comment by f-f** [comment]
I read this through, and it looks like to support the whole flow of auto-autentication for all the environments we need to pull in the SDK for one of the supported languages (Java in this case), which I'm not sure it's a dependency we want to depend on (as it is a client on its own already).

On the bright side, supporting the reading of the service account key from the path provided in the env variable `GOOGLE_APPLICATION_CREDENTIALS` should be straightforward to implement.

**Comment by f-f** [comment]
@zudov given the previous comment, is fc701a7 enough to consider this closed?

**Comment by zudov** [comment]
Yep. Just a variable would do good for now. Doesn't feel good to pull the java SDK dependency for just that (given that otherwise we can be free of it). So maybe when the time is good we just implement the full "Application default credentials" in clojure and have it in another library. They've got implementations in quite many languages there, so shouldn't be hard to figure it out.

Dunno if we should close this issue. It's "solved" for now, but there's more work to be done. We can either keep it open, or close it and reopen later, or put it into some "production ready" milestone.

**Comment by f-f** [comment]
Sounds good, implementing the full "Application Default Credentials" thing in Clojure shouldn't be too complicated, porting [their implementation from Java](https://github.com/google/google-api-java-client/blob/dev/google-api-client/src/main/java/com/google/api/client/googleapis/auth/oauth2/DefaultCredentialProvider.java#L107) looks quite feasible.

Let's have this issue open, but add a label for denoting the "not a priority" status.
